### PR TITLE
[cu-206y1te] WIP: Make Vesting Pallet Instantiable to Support Timestamp based Vesting

### PR DESCRIPTION
**Pull request by vimukthi-git**
Make Vesting Pallet Instantiable to Support Timestamp based Vesting

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-206y1te Change vesting schedule to be able to use absolute time in addition block number](https://app.clickup.com/t/206y1te)

File: frame/vesting/src/lib.rs [hunk 1]
```diff
@@ -154,30 +163,30 @@ pub mod module {
 	// FIXME: Temporary fix to get CI to pass, separate PRs will be made per pallet to refactor to
 	// use OptionQuery instead
 	#[allow(clippy::disallowed_type)]
-	pub type VestingSchedules<T: Config> = StorageDoubleMap<
+	pub type VestingSchedules<T: Config<I>, I: 'static = ()> = StorageDoubleMap<
```
Comment (vimukthi-git) [inline frame/vesting/src/lib.rs:166]: Most compilation issues are related to this as of now...

File: frame/vesting/src/mock.rs [hunk 2]
```diff
@@ -172,15 +218,15 @@ impl ExtBuilder {
...
-		vesting::GenesisConfig::<Runtime> {
-			vesting: vec![
-				// asset, who, start, period, period_count, per_period
-				(MockCurrencyId::BTC, CHARLIE, 2, 3, 1, 5),
-				(MockCurrencyId::BTC, CHARLIE, 2 + 3, 3, 3, 5),
-			],
-		}
-		.assimilate_storage(&mut t)
-		.unwrap();
+		// vesting::GenesisConfig::<Runtime> {
```
Comment (vimukthi-git) [inline frame/vesting/src/mock.rs:221]: Commented out temporarily

File: frame/vesting/Cargo.toml [hunk 3]
```diff
@@ -23,8 +23,9 @@ composable-traits = { version = "0.0.1", path = "../../frame/composable-traits",
 orml-traits = { git = "https://github.com/open-web3-stack/open-runtime-module-library", rev = "2b1c9fb367ccb8e13601b2da43d1c5d9737b93c6", default-features = false }
 
 [dev-dependencies]
+pallet-timestamp = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.16" }
 sp-core = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.16" }
-orml-tokens = { git = "https://github.com/open-web3-stack/open-runtime-module-library", rev = "2b1c9fb367ccb8e13601b2da43d1c5d9737b93c6", default-features = false }
+orml-tokens = { git = "https://github.com/open-web3-stack/open-runtime-module-library", rev = "2b1c9fb367ccb8e13601b2da43d1c5d9737b93c6", default-features = true }
```
Comment (vimukthi-git) [inline frame/vesting/Cargo.toml:28]: Why was this set to false?

Comment (hussein-aitlahcen) [reply frame/vesting/Cargo.toml:28]: Probably an artifact

**Pull request closed without merging**